aggregator/aggregator_test.go

```diff
@@ -560,6 +560,45 @@ func TestAggregatorOwnedShards(t *testing.T) {
 	}
 }
 
+func TestAggregatorAddMetricMetrics(t *testing.T) {
+	s := tally.NewTestScope("testScope", nil)
+	m := newAggregatorAddMetricMetrics(s, 1.0)
+	m.ReportSuccess(time.Second)
+	m.ReportError(errWriteNewMetricRateLimitExceeded)
+	m.ReportError(errWriteValueRateLimitExceeded)
+	m.ReportError(errors.New("foo"))
+
+	snapshot := s.Snapshot()
+	counters, timers, gauges := snapshot.Counters(), snapshot.Timers(), snapshot.Gauges()
+
+	// Validate we count successes and errors correctly.
+	require.Equal(t, 4, len(counters))
+	for _, id := range []string{
+		"testScope.success+", // NB: this looks like a bug with tally test scope
+		"testScope.errors+reason=value-rate-limit-exceeded",
+		"testScope.errors+reason=new-metric-rate-limit-exceeded",
+		"testScope.errors+reason=not-categorized",
+	} {
+		c, exists := counters[id]
+		require.True(t, exists)
+		require.Equal(t, int64(1), c.Value())
+	}
+
+	// Validate we record times correctly.
+	require.Equal(t, 1, len(timers))
+
+	for _, id := range []string{
+		"testScope.success-latency+", // NB: this looks like a bug with tally test scope
```

Comment:
We talked about this in the past: https://github.com/uber-go/tally/pull/41 I think the decision was to not break backwards compatibility.